src/libint_3center.F

```diff
@@ -0,0 +1,319 @@
+!> \brief 3-center electron repulsion integral routines based on libint2
+!>        Currently available operators: Coulomb, Truncated Coulomb, Short Range (erfc)
+!> \author A. Bussy (05.2019)
+! **************************************************************************************************
+
+MODULE libint_3center
+   USE gamma,                           ONLY: fgamma => fgamma_0
+   USE hfx_libint_wrapper,              ONLY: cp_libint_get_3eris,&
+                                              cp_libint_set_params_eri,&
+                                              cp_libint_t,&
+                                              prim_data_f_size
+   USE input_constants,                 ONLY: do_potential_coulomb,&
+                                              do_potential_short,&
+                                              do_potential_truncated
+   USE kinds,                           ONLY: dp
+   USE mathconstants,                   ONLY: pi
+   USE orbital_pointers,                ONLY: nco,&
+                                              ncoset
+   USE t_c_g0,                          ONLY: get_lmax_init,&
+                                              t_c_g0_n
+#include "./base/base_uses.f90"
+
+   IMPLICIT NONE
+   PRIVATE
+
+   CHARACTER(len=*), PARAMETER, PRIVATE :: moduleN = 'libint_3center'
+
+   PUBLIC :: eri_3center
+
+   TYPE :: params_3c
+      INTEGER                               :: m_max
+      REAL(dp)                              :: ZetaInv, EtaInv, ZetapEtaInv, Rho
+      REAL(dp), DIMENSION(3)                :: Q, W
+      REAL(dp), DIMENSION(prim_data_f_size) :: Fm
+   END TYPE
+
+CONTAINS
+
+! **************************************************************************************************
+!> \brief Computes the 3-center electron repulsion integrals (ab|c) for a given set of cartesian
+!>        gaussian orbitals
+!> \param int_abc the integrals as array of cartesian orbitals (allocated before hand)
+!> \param la_min ...
+!> \param la_max ...
+!> \param npgfa ...
+!> \param zeta ...
+!> \param ra ...
+!> \param lb_min ...
+!> \param lb_max ...
+!> \param npgfb ...
+!> \param zetb ...
+!> \param rb ...
+!> \param lc_min ...
+!> \param lc_max ...
+!> \param npgfc ...
+!> \param zetc ...
+!> \param rc ...
+!> \param lib the libint_t object for evaluation (assume that it is initialized outside)
+!> \param op which operator to use for the integrals
+!> \param omega in case of short range erfc operator
+!> \param r_cutoff the cutoff radius in case of truncated Coulomb
+!> \note Prior to calling this routine, the cp_libint_t type passed as argument must be initialized,
+!>       the libint library must be static initialized, and in case of truncated Coulomb operator,
+!>       the latter must be initialized too
+! **************************************************************************************************
+   SUBROUTINE eri_3center(int_abc, la_min, la_max, npgfa, zeta, ra, &
```

Comment:
For performance, you should apply screening to each pair of primitive Gaussians, for sure in a,b but also b,c and a,c in case of a truncated/short-range operator (see for instance [overlap3](https://github.com/cp2k/cp2k/blob/master/src/aobasis/ai_overlap3.F#L173-L188) subroutine).

Reply:
OK, but let's leave it like this for now as we discussed in real life, until you push your improved version of it.